tapestry.js

```diff
@@ -50,16 +53,48 @@ $.getJSON( jsonUrl, function(result){
     
     if (saveProgressToCookie) {
         // Update dataset with data from cookie (if any)
-        var cookieProgress = Cookies.get("progress-data-"+tapestrySlug);
-        if (cookieProgress !== undefined) {
-            cookieProgress = JSON.parse( cookieProgress );
-            setDatasetProgress(cookieProgress);
-        }
-        // Update H5P Video Settings from cookie (if any)
-        var cookieH5PVideoSettings = Cookies.get("h5p-video-settings");
-        if (cookieH5PVideoSettings !== undefined) {
-            cookieH5PVideoSettings = JSON.parse( cookieH5PVideoSettings );
-            h5pVideoSettings = cookieH5PVideoSettings;
+        var cookieProgress; 
+        var cookieH5PVideoSettings;
+
+        // If user is logged in, get progress from database database
+        if (tapestryWpUserId) {
+            var nodeIdArr = dataset.nodes.map(node => node.id);
```

Comment:
Use [`array_map` function](http://docs.php.net/manual/da/function.array-map.php) instead since ES6 might not be supported in some browsers (see [here](https://www.w3schools.com/js/js_versions.asp) for more info)